pkg/installer/installer.go

```diff
@@ -442,3 +443,43 @@ func (in *Installer) getBackupPath() (string, error) {
 	}
 	return filepath.Join(homeDir, kubeDir, stosDir, fmt.Sprintf("%s%v", uninstallDirPrefix, in.kubeClusterID)), nil
 }
+
+func (in *Installer) getDecodedAPISecretUsername(storageosAPISecret string) (string, error) {
+	secretUsername, err := pluginutils.GetFieldInManifest(storageosAPISecret, "data", "username")
+	if err != nil {
+		return "", err
+	}
+	if secretUsername == "" {
+		// also check for apiUsername (pre 2.5.0)
+		secretUsername, err = pluginutils.GetFieldInManifest(storageosAPISecret, "data", "apiUsername")
+		if err != nil {
+			return "", err
+		}
+	}
+	decodedSecretUsername, err := base64.StdEncoding.DecodeString(secretUsername)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	return string(decodedSecretUsername), nil
+}
+
+func (in *Installer) getDecodedAPISecretPassword(storageosAPISecret string) (string, error) {
```

Comment:
I don't see much difference between `getDecodedAPISecretUsername` and `getDecodedAPISecretPassword`. Is there any?

Reply:
You're right, I was thinking of a nice way to generalise this behaviour. I'll make a ticket to come back and do this 😅